modules/dicom_archive/php/viewdetails.class.inc

```diff
@@ -51,7 +51,47 @@ class ViewDetails extends \NDB_Form
      */
     function _hasAccess(\User $user) : bool
     {
-        return $user->hasPermission('dicom_archive_view_allsites');
+        // remove the possibility to have no tarchive ID in this page
+        if (empty($_REQUEST['tarchiveID'])) {
+            // defaults to permission denied
+            return false;
+        }
+
+        // get project ID from Tarchive ID.
+        $tarchiveID = intval($_REQUEST['tarchiveID']);
```

Comment:
You shouldn't be using the $_REQUEST superglobal. hasAccess can be called from non-web contexts. (And all superglobals are generally bad practice.)

It looks like this class already has a public $tarchiveID attribute that you should be using. If it's not initialized when this is called, that's what the loadResources function (which is called before hasAccess in a web context) is for.

Reply:
Oh, I did not know about `loadResources`, thanks I will check that.